vm/actor/src/builtin/miner/mod.rs

```diff
@@ -1014,185 +1063,111 @@ impl Actor {
                     .iter()
                     .chain(&[info.worker, info.owner]),
             )?;
-
-            if consensus_fault_active(&info, rt.curr_epoch()) {
-                return Err(actor_error!(
-                    ErrForbidden,
-                    "precommit not allowed during active consensus fault"
-                ));
-            }
-
-            // From network version 7, the pre-commit seal type must have the same Window PoSt proof type as the miner's
-            // recorded seal type has, rather than be exactly the same seal type.
-            // This permits a transition window from V1 to V1_1 seal types (which share Window PoSt proof type).
-            let sector_wpost_proof =
-                params
-                    .seal_proof
-                    .registered_window_post_proof()
-                    .map_err(|e| {
-                        actor_error!(
-                            ErrIllegalState,
-                            "failed to lookup window PoSt proof type \
-                            for sector seal proof {:?}: {}",
-                            params.seal_proof,
-                            e
-                        )
-                    })?;
-            if sector_wpost_proof != info.window_post_proof_type {
-                return Err(actor_error!(
-                    ErrIllegalArgument,
-                    "sector window PoSt proof type {:?} must match miner \
-                        window PoSt proof type {:?}",
-                    sector_wpost_proof,
-                    info.window_post_proof_type
-                ));
-            }
-
             let store = rt.store();
-
-            let deal_count_max = sector_deals_max(info.sector_size);
-            if params.deal_ids.len() as u64 > deal_count_max {
-                return Err(actor_error!(
-                    ErrIllegalArgument,
-                    "too many deals for sector {} > {}",
-                    params.deal_ids.len(),
-                    deal_count_max
-                ));
+            let info = get_miner_info(store, state)?;
```

Comment:
This call to `get_miner_info` seems redundant because it doesn't seem like `info` gets consumed when first used. Do we have to keep this because Lotus also redundantly retrieves the miner info?

Reply:
Fixed!